packages/payment-processor/src/payment/utils.ts

```diff
@@ -28,8 +28,8 @@ export class UnsupportedCurrencyNetwork extends Error {
  * Utility to get the default window.ethereum provider, or throws an error.
  */
 export function getProvider(): providers.Web3Provider {
-  if (typeof window !== 'undefined' && 'ethereum' in window) {
-    return new ethers.providers.Web3Provider((window as any).ethereum);
+  if ('ethereum' in globalThis) {
+    return new ethers.providers.Web3Provider((globalThis as any).ethereum);
   }
```

Comment:
Without this change, we would get the following error during build:

```
✖  @requestnetwork/payment-processor:build
$ tsc -b tsconfig.build.json
       src/payment/utils.ts(31,14): error TS2304: Cannot find name 'window'.
```

I'm guessing it worked before because `graphql-request` v6 declared a `window` global type.